api/client/login.go

```diff
@@ -74,6 +74,10 @@ func (cli *DockerCli) CmdLogin(args ...string) error {
 			username = authconfig.Username
 		}
 	}
+
+	if username == "" {
+		return fmt.Errorf("Error: Non-null Username Required")
+	}
```

Comment:
well, if username before line 69 was "    " then it won't be trimmed, right?  And if authconfig.Username in the json had extra spaces then it won't be trimmed.